src/compiler/crystal/tools/formatter.cr

```diff
@@ -220,9 +220,14 @@ module Crystal
         write "begin"
         @indent += 2
         write_line
-        next_token_skip_space_or_newline
-        if @token.type == :";"
-          next_token_skip_space_or_newline
+        next_token
+        # Cornor case: an empty `begin ... end`.
+        # In this case, we should not skip space becuase it will do in the below loop.
```

Comment:
```suggestion
        # Corner case: an empty `begin ... end`.
        # In this case, we should not skip space because it will do in the below loop.
```